core/src/main/java/org/mskcc/cbio/portal/scripts/ImportGenericAssayEntity.java

```diff
@@ -112,75 +127,132 @@ public void run() {
     *
     * @param updateInfo
     */
-    public static void startImport(OptionSet options, OptionSpec<String> data, boolean updateInfo) throws Exception {
-        if (options.hasArgument(data)) {
+    public static void startImport(OptionSet options, OptionSpec<String> data, OptionSpec<String> geneticAlterationType, boolean updateInfo) throws Exception {
+        if (options.hasArgument(data) && options.hasArgument(geneticAlterationType)) {
             File treatmentFile = new File(options.valueOf(data));
-            importData(treatmentFile);
+            GeneticAlterationType geneticAlterationTypeArg = GeneticAlterationType.valueOf(options.valueOf(geneticAlterationType));
+            importData(treatmentFile, geneticAlterationTypeArg);
         }
     }
     
     /**
     * Imports feature columns from treatment file.
     *
-    * @param treatmentDataFile
+    * @param dataFile
     * @param updateInfo boolean that indicates wether additional fields
     * ("Description, Name, URL") of existing records should be overwritten
     * @throws Exception
     */
-    public static void importData(File treatmentDataFile) throws Exception {
+    public static void importData(File dataFile, GeneticAlterationType geneticAlterationType) throws Exception {
         
-        ProgressMonitor.setCurrentMessage("Reading data from: " + treatmentDataFile.getCanonicalPath());
+        ProgressMonitor.setCurrentMessage("Reading data from: " + dataFile.getCanonicalPath());
         
         // read gene set data file - note: this file does not contain headers
-        FileReader reader = new FileReader(treatmentDataFile);
+        FileReader reader = new FileReader(dataFile);
         BufferedReader buf = new BufferedReader(reader);
         String currentLine = buf.readLine();
         String[] headerNames = currentLine.split("\t");
         
-        int indexStableIdField = getTreatmentStableIdIndex(headerNames);
-        int indexNameField = getTreatmentNameIndex(headerNames);
-        int indexDescField = getTreatmentDescIndex(headerNames);
-        int indexUrlField = getTreatmentUrlIndex(headerNames);
-        
-        currentLine = buf.readLine();
-        
-        while (currentLine != null) {
+        if (geneticAlterationType == GeneticAlterationType.MUTATIONAL_SIGNATURE) {
+            // read generic assay data
+            int indexStableIdField = getStableIdIndex(headerNames);
```

Comment:
In a parallel discussion about [PR-6457](https://github.com/cBioPortal/cbioportal/pull/6457), we decided to add a property to the meta_ file which lists the columns in the datafile to be considered meta-properties.  For consistency, we should do the same here.  We can discuss this further in person.